ios/FluentUI/DrawerVnext/Drawer.swift

```diff
@@ -0,0 +1,170 @@
+    /// If set to `false` it restores to `clear` color
+    @objc @Published public var backgroundDimmed: Bool = false
+}
+
+public typealias DrawerStateChangedCompletionBlock = (_ isExpanded: Bool) -> Void
+
+// MARK: - Drawer Token
+
+/// `DrawerTokens` assist to configure drawer apperance via UIKit components.
+public class DrawerTokens: ObservableObject {
+
+    @Published public var shadowColor: Color!
+    @Published public var shadowOpacity: Double!
+    @Published public var shadowBlur: CGFloat!
+    @Published public var shadowDepthX: CGFloat!
+    @Published public var shadowDepthY: CGFloat!
+
+    public init() {
+        self.themeAware = true
+        didChangeAppearanceProxy()
+    }
+
+    @objc open func didChangeAppearanceProxy() {
+        let appearanceProxy = StylesheetManager.S.DrawerTokens
+        shadowColor = Color(appearanceProxy.shadowColor)
+        shadowOpacity = Double(appearanceProxy.shadowOpacity)
+        shadowBlur = appearanceProxy.shadowBlur
+        shadowDepthX = appearanceProxy.shadowX
+        shadowDepthY = appearanceProxy.shadowY
+    }
+}
+
+// MARK: - Drawer
+
+/// `Drawer` is used to present a overlay a content partially on another view.
+/// `Drawer`  support horizontal axis and is expanded by default from left side of the screen unless explicitly specified
+///  Set `Content` to provide content for the drawer.
+public struct Drawer<Content: View>: View {
+
+    // content view on top of `Drawer`
+    public var content: Content
+
+    // configure the behavior of drawer
+    @ObservedObject public var state = DrawerState()
+
+    // configure the apperance of drawer
+    @ObservedObject public var tokens = DrawerTokens()
+
+    // keep track of dragged offset
+    @State internal var draggedOffsetWidth: CGFloat?
+
+    // internal drawer state
+    @State internal var isDrawerOpen: Bool = false
+
+    public var body: some View {
+        GeometryReader { proxy in
+            SlideOverPanel(
+                content: content,
+                isOpen: $isDrawerOpen,
+                preferredContentOffset: $draggedOffsetWidth)
+                .backgroundOpactiy(backgroundLayerOpacity)
+                .direction(slideOutDirection)
+                .width(proxy.portraitOrientationAgnosticSize().width)
+                .performOnBackgroundTap {
+                    state.isExpanded.toggle()
+                }
+                .onReceive(state.$isExpanded, perform: { value in
+                    withAnimation {
+                        self.isDrawerOpen = value
+                        self.draggedOffsetWidth = nil // drag ends
+                    }
+                })
+                .gesture(dragGesture(screenWidth: proxy.portraitOrientationAgnosticSize().width))
+        }
+        .edgesIgnoringSafeArea(.all)
+    }
+
+    private var backgroundLayerOpacity: Double {
+        return state.backgroundDimmed ? 0.5 : 0
```

Comment:
This opacity value should be tokenized. Can you start a conversation with Victor about it? (and open a GitHub issue for it)